server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -2324,7 +2339,7 @@ private void validateVolumeResizeWithSize(VolumeVO volume, long currentSize, Lon
              * the actual disk size.
              */
             if (currentSize > newSize) {
-                if (volume != null && ImageFormat.QCOW2.equals(volume.getFormat()) && !Volume.State.Allocated.equals(volume.getState())) {
+                if (volume != null && ImageFormat.QCOW2.equals(volume.getFormat()) && !Volume.State.Allocated.equals(volume.getState()) && !StoragePoolType.StorPool.equals(volume.getPoolType())) {
```

Comment:
hard-coded check, can check this through StorPool storage plugin?

Reply:
@sureshanaparti, there are a lot similar checks in the code 
https://github.com/apache/cloudstack/blob/6dc3d06037c39019f29686281856443c37a3e6c0/server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java#L3229

I think there is no need to complicate the code by adding additional method in the storage drivers

Reply:
agree @slavkap , maybe these hard-coded checks have to be fixed at some point of time with proper testing.